src/controllers/activegate/reconciler/statefulset/statefulset.go

```diff
@@ -333,33 +338,38 @@ func buildEnvs(stsProperties *statefulSetProperties) []corev1.EnvVar {
 	deploymentMetadata := deploymentmetadata.NewDeploymentMetadata(string(stsProperties.kubeSystemUID), deploymentmetadata.DeploymentTypeActiveGate)
 
 	envs := []corev1.EnvVar{
-		{
-			Name: dtServer,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: stsProperties.AGTenantSecret(),
+		{Name: dtCapabilities, Value: stsProperties.capabilityName},
+		{Name: dtIdSeedNamespace, Value: stsProperties.Namespace},
+		{Name: dtIdSeedClusterId, Value: string(stsProperties.kubeSystemUID)},
+		{Name: dtDeploymentMetadata, Value: deploymentMetadata.AsString()},
+	}
+
+	if stsProperties.DynaKube.FeatureEnableActivegateRawImage() {
+		envs = append(envs,
+			corev1.EnvVar{
+				Name: dtServer,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: stsProperties.AGTenantSecret(),
+						},
+						Key: activegate.CommunicationEndpointsName,
 					},
-					Key: activegate.CommunicationEndpointsName,
 				},
 			},
-		},
-		{
-			Name: dtTenant,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: stsProperties.AGTenantSecret(),
+			corev1.EnvVar{
+				Name: dtTenant,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: stsProperties.AGTenantSecret(),
+						},
+						Key: activegate.TenantUuidName,
 					},
-					Key: activegate.TenantUuidName,
 				},
```

Comment:
I'd suggest creating functions to build the env vars. E.g., having to functions `communicationEndpointEnvVar()` and `tenantUuuidName()`.